element-hq/element-x-android
# Fix verification failed issue, simplify verification logic

## Content

- Avoid unnecessary `initVerificationControllerIfNeeded` call which caused the automatic verification flow failure.
- Reuse Rust `Client` instances created on the login process so we don't need to restore one right before the session verification.
- Remove unnecessary sources of verification state updates.
- Add an intermediate FTUE flow step which will display an indeterminate progress indicator instead of a blank screen.

## Motivation and context

This should fix the issues we've seen where the authentication failed automatically due to a race condition.

Keeping the Client instances created on login would also make our setup more similar to iOS and avoid unnecessary work.

## Tests

- Login and verify your device (self-verification).
- Try verifying some other device from this one.

## Tested devices

- OS version(s): 14

## Checklist
